crates/client/eth/src/client.rs

```diff
@@ -0,0 +1,174 @@
+use std::sync::Arc;
+
+use alloy::sol_types::SolEvent;
+use alloy::{
+    primitives::Address,
+    providers::{Provider, ProviderBuilder, ReqwestProvider, RootProvider},
+    rpc::types::Filter,
+    sol,
+    transports::http::{Client, Http},
+};
+use anyhow::{bail, Context};
+use bitvec::macros::internal::funty::Fundamental;
+use starknet_api::hash::StarkFelt;
+use url::Url;
+
+use crate::client::StarknetCoreContract::StarknetCoreContractInstance;
+use crate::{config::L1StateUpdate, utils::u256_to_starkfelt};
```

Comment:
we should move `L1StateUpdate` to `state_update.rs` and with the `get_initial_state` function

Reply:
refactored